presto-main/src/main/java/com/facebook/presto/execution/scheduler/SimpleNodeSelector.java

```diff
@@ -168,6 +171,19 @@ else if (!splitWaitingForAnyNode) {
         return new SplitPlacementResult(blocked, assignment);
     }
 
+    private ResettableRandomizedIterator<InternalNode> getRandomCandidates(int limit, NodeMap nodeMap, List<RemoteTask> existingTasks)
+    {
+        Set<InternalNode> existingNodes = existingTasks.stream().map(remoteTask -> nodeMap.getNodesByNodeId().get(remoteTask.getNodeId())).collect(toSet());
```

Comment:
suggestion:

This code is gonna be called for every batch of splits. We can make it more efficient by

1. Collecting it here to a `List`
2. Create a set only in the less "likely" branch: `alreadySelectedNodeCount < limit`, right before it has to be passed to the `selectNodes`

Reply:
Agreed with @arhimondr . Calling it per each `computeAssignments` call might be expensive. 

Also, we usually have new lines for stream API call:
```java
existingTasks.stream()
      .map(remoteTask -> nodeMap.getNodesByNodeId().get(remoteTask.getNodeId()))
      .collect(toSet());
```